90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -219,6 +219,40 @@ clone_snapshot() {
   return 0
 }
 
+# arg1: ZFS filesystem
+# arg2: default kernel path (omit to unset default)
+# prints: nothing
+# returns: 0 on success, 1 otherwise
+
+set_default_kernel() {
+  local fs kernel
+
+  fs="$1"
+  [ -n "${fs}" ] || return 1
+
+  pool="${fs%%/*}"
+  [ -n "${pool}" ] || return 1
+
+  # Strip /boot/ to list only the file
+  kernel="${2#/boot/}"
+
+  # Make sure the pool is writable
+  if set_rw_pool "${pool}"; then
+    key_wrapper "${pool}"
```

Comment:
`key_wrapper` should have  `CLEAR_SCREEN=1` set before it's called, and `CLEAR_SCREEN=0` set after. If this isn't done, the password input prompt appears on the bottom of the screen, where fzf leaves the cursor for input.

This is a giant hack, but a global variable was the only reasonable way to make this happen.